speechbrain/lobes/models/discrete/sq_codec.py

```diff
@@ -0,0 +1,1394 @@
+        x : torch.Tensor
+            Input tensor.
+
+        Returns
+        -------
+        torch.Tensor
+            Processed output tensor.
+        """
+        for conv in self.convs:
+            x = conv(x)
+        x = self.down_conv(x)
+        return x
+
+
+class ResDecoderBlock(nn.Module):
+    """
+    A residual decoder block with upsampling and multiple residual units.
+
+    Arguments
+    ---------
+    n_in : int
+        Number of input channels.
+    n_out : int
+        Number of output channels.
+    stride : int
+        Stride for the upsampling layer.
+    up_kernel_size : int
+        Kernel size for the upsampling layer.
+    res_kernel_size : int, optional
+        Size of the convolutional kernel for residual connections (default is 7).
+    causal : bool, optional
+        If True, applies causal convolution (default is False).
+    """
+
+    def __init__(
+        self,
+        n_in,
+        n_out,
+        stride,
+        up_kernel_size,
+        res_kernel_size=7,
+        causal=False,
+    ):
+        super(ResDecoderBlock, self).__init__()
+        self.up_conv = UpsampleLayer(
+            n_in,
+            n_out,
+            kernel_size=up_kernel_size,
+            stride=stride,
+            causal=causal,
+            activation=None,
+        )
+        self.convs = nn.ModuleList(
+            [
+                ResidualUnit(
+                    n_out,
+                    n_out,
+                    dilation=1,
+                    res_kernel_size=res_kernel_size,
+                    causal=causal,
+                ),
+                ResidualUnit(
+                    n_out,
+                    n_out,
+                    dilation=3,
+                    res_kernel_size=res_kernel_size,
+                    causal=causal,
+                ),
+                ResidualUnit(
+                    n_out,
+                    n_out,
+                    dilation=5,
+                    res_kernel_size=res_kernel_size,
+                    causal=causal,
+                ),
+                ResidualUnit(
+                    n_out,
+                    n_out,
+                    dilation=7,
+                    res_kernel_size=res_kernel_size,
+                    causal=causal,
+                ),
+                ResidualUnit(
+                    n_out,
+                    n_out,
+                    dilation=9,
+                    res_kernel_size=res_kernel_size,
+                    causal=causal,
+                ),
+            ]
+        )
+
+    def forward(self, x):
+        """
+        Applies upsampling followed by a series of residual units.
+
+        Arguments
+        ---------
+        x : torch.Tensor
+            Input tensor.
+
+        Returns
+        -------
+        torch.Tensor
+            Processed output tensor.
+        """
+        x = self.up_conv(x)
+        for conv in self.convs:
+            x = conv(x)
+        return x
+
+
+class Conv1d(nn.Conv1d):
```

Comment:
If I use the SB downsampling, it causes an error when loading the pretrained weights since the naming could be a bit different.  Since it is going to be moved to the integration folder, I suggest keep all their module as it is . 
I am working on another recepie that include many of these quantizati